### PR TITLE
Add position as a parameter for TemplateSpatialModel

### PR DESCRIPTION
**Pull request by AtreyeeS**
This PR allows the fitting of the position for template spatial models, and addresses #4920

### CODE REVIEW

**Review by QRemy — COMMENTED** [review]
Thanks @AtreyeeS, could you fix the failing test and add you to check if we can read a yaml model where the position is not defined ?

**Comment by AtreyeeS** [comment]
Thanks @QRemy ! Indeed the serialisation was broken.
I have added a test for the to and from dict now.

**Review by QRemy — CHANGES_REQUESTED** [review]
I think it is possible to have it work without addins the parameters argument on init :

File: gammapy/modeling/models/spatial.py [hunk 1]
```diff
@@ -1325,8 +1327,36 @@ def __init__(
...
         super().__init__(**kwargs)
 
+    def __str__(self):
+        width = self.map.geom.width
+        data_min = np.nanmin(self.map.data)
+        data_max = np.nanmax(self.map.data)
+
+        prnt = (
+            f"{self.__class__.__name__} model summary:\n "
+            f"Reference position: {self.position} \n "
```
Comment (adonath) [inline gammapy/modeling/models/spatial.py:1344]: I think this is confusing. The WCS reference position doe snot change and is not equivalent to `self.position`.

File: gammapy/modeling/models/spatial.py [hunk 2]
```diff
@@ -1325,8 +1327,36 @@ def __init__(
...
```
Comment (AtreyeeS) [inline gammapy/modeling/models/spatial.py:1344]:
```suggestion
            f"Model center: {self.position} \n "
```

File: gammapy/modeling/models/spatial.py (hunk 1 above)
Comment (AtreyeeS) [reply gammapy/modeling/models/spatial.py]: @adonath The reference was a bad naming. ( I was following the `LightCurveTemplateTemporalModel`, where it makes more sense). Updated to `Model position`

**Review by QRemy — DISMISSED** [review]
Thanks @AtreyeeS, no further comment from my side.

**Review by registerrier — APPROVED** [review]
Thanks @AtreyeeS . This is nice. 

The way the coordinate rotation is applied should be explained with a caveat on the applicability. Or conversely one can determine the rotation angles and apply them to the coords.

File: gammapy/estimators/map/tests/test_ts.py [hunk 3]
```diff
@@ -341,3 +342,21 @@ def test_MapDatasetOnOff_error():
     ts_estimator = TSMapEstimator()
     with pytest.raises(TypeError):
         ts_estimator.run(dataset=dataset_on_off)
+
+
+@requires_data()
+def test_with_TemplateSpatialModel():
```
[the remark this answers is not in the document]
Comment (AtreyeeS) [reply gammapy/estimators/map/tests/test_ts.py:348]: I downsampled the dataset. Now it takes 2s

**Review by QRemy — APPROVED** [review]
Thanks @AtreyeeS, merging now

**Pull request merged**